x-pack/plugins/security_solution/common/endpoint/types/index.ts

```diff
@@ -61,6 +61,11 @@ type ImmutableMap<K, V> = ReadonlyMap<Immutable<K>, Immutable<V>>;
 type ImmutableSet<T> = ReadonlySet<Immutable<T>>;
 type ImmutableObject<T> = { readonly [K in keyof T]: Immutable<T[K]> };
 
+/**
+ * Utility type that will return back a union of the given [T]ype and an Immutable version of it
+ */
+export type MaybeImmutable<T> = T | Immutable<T>;
```

Comment:
@parkiino @pzl FYI. Added this generic utility type because (at least in client side code) we continue to define types (especially as input arguments) to accept a mutable and immutable type.